File: ts/src/optimization/DateTimeUtil.ts
function formatDate(date: Date): string {
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, "0");
  const day = String(date.getDate()).padStart(2, "0");
  const hours = String(date.getHours()).padStart(2, "0");
  const minutes = String(date.getMinutes()).padStart(2, "0");
  const seconds = String(date.getSeconds()).padStart(2, "0");

  return `${year}-${month}-${day}_${hours}:${minutes}:${seconds}`;
}

function formatDurationToSec(start: Date, end: Date): string {
  const durationMs = end.getTime() - start.getTime();
  const durationSec = durationMs / 1000;
  return durationSec.toFixed(3) + "s";
}

const DateTimeUtil = Object.freeze({ formatDate, formatDurationToSec });
export default DateTimeUtil;

File: java/src/main/java/com/cc/optimization/DateTimeUtil.java
package com.cc.optimization;

import java.util.Date;

public class DateTimeUtil {
    public static String formatDate(Date date) {
        int year = date.getYear() + 1900;
        String month = String.format("%02d", date.getMonth() + 1);
        String day = String.format("%02d", date.getDate());
        String hours = String.format("%02d", date.getHours());
        String minutes = String.format("%02d", date.getMinutes());
        String seconds = String.format("%02d", date.getSeconds());

        return String.format("%d-%s-%s_%s:%s:%s", year, month, day, hours, minutes, seconds);
    }

    public static String formatDurationToSec(Date start, Date end) {
        long durationMs = end.getTime() - start.getTime();
        double durationSec = durationMs / 1000.0;
        return String.format("%.3fs", durationSec);
    }
}
